temporalio/lib/temporalio/worker/poller_behavior.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Temporalio
+  class Worker
+    # Base class for poller behaviors that control how polling scales.
+    #
+    # Use factory methods {.simple_maximum} or {.autoscaling} to create instances.
+    class PollerBehavior
+      # Creates a simple maximum poller behavior
+      # The poller will attempt to poll as long as a slot is available, up to the
+      # provided maximum. Cannot be less than two for workflow tasks, or one for other tasks.
+      #
+      # @param maximum [Integer] Maximum number of concurrent poll requests.
+      # @return [SimpleMaximumPollerBehavior] A simple maximum poller behavior
+      def self.simple_maximum(maximum = 5)
+        SimpleMaximumPollerBehavior.new(maximum: maximum)
+      end
+
+      # Creates an autoscaling poller behavior
+      # The poller will automatically scale the number of pollers based on feedback
+      # from the server. A slot must be available before beginning polling.
+      #
+      # @param minimum [Integer] Minimum number of poll calls (assuming slots are available).
+      # @param maximum [Integer] Maximum number of poll calls that will ever be open at once.
+      # @param initial [Integer] Number of polls attempted initially before scaling kicks in.
+      # @return [AutoscalingPollerBehavior] An autoscaling poller behavior
+      def self.autoscaling(minimum: 1, maximum: 100, initial: 5)
+        AutoscalingPollerBehavior.new(minimum: minimum, maximum: maximum, initial: initial)
+      end
+
+      # @!visibility private
+      def _to_bridge_options
+        raise NotImplementedError, 'Subclasses must implement this method'
+      end
+    end
+
+    # A poller behavior that attempts to poll as long as a slot is available, up to the
+    # provided maximum. Cannot be less than two for workflow tasks, or one for other tasks.
+    class SimpleMaximumPollerBehavior < PollerBehavior
```

Comment:
```suggestion
    class SimpleMaximum < PollerBehavior
```

It is idiomatic in Ruby to remove this suffix (same for autoscaling one below)

Reply:
If so it seems like these should be inside a `PollerBehavior` module?